docs/backup_and_recovery.md

```diff
@@ -0,0 +1,52 @@
+# Client E-cash backup&recovery
+
+## Introduction
+
+The client uses [the deterministic e-cash derivation scheme described in](./recoverable_e-cash.md). This allows e-cash recovery from the root key (exported and backed up securely during wallet creation) in case of data corruption or losing access to the wallet.
+
+Using the root key it's possible to deterministically derive all the secret material, and by matching it against publically available Fedimint federation history, recreate all the lost data.
+
+However, recreating user funds by scanning the entire federation history might be very time and resource-consuming. To help with this problem Fedimint's backup&recovery supports backup snapshots, allowing practical and fast, yet privacy-preserving recovery.
+
+This document is only a high-level description and might be out of date. For details, please refer to the source code itself, which contains a lot of nitty-gritty information.
+
+## Creating backup snapshot
+
+Periodically, on a fixed schedule (for privacy), the client should create and upload a backup snapshot of their e-cash data.
+
+The following is the persistent data relevant to the e-cash derivation:
+
+* spendable signed notes
+* unconfirmed notes
+* note derivation index counter for each amount tier
+
+An atomic snapshot of all the above data is taken, along with the current Fedimint consensus `epoch`. The embedded `epoch` is what allows skipping a large part of Federation history during the restoration procedure.
+
+All this data is serialized, padded (to avoid leaking any information), and encrypted with the deterministically derived (from the root seed key) backup encryption key.
```

Comment:
Would be good to add a section for the derivation path here https://github.com/fedimint/fedimint/blob/f3ee5b8e382ed59702227a9a9c7f6be3fcd9ea46/docs/recoverable_e-cash.md#derivation-paths